http.go

```diff
@@ -0,0 +1,87 @@
+// Copyright 2023 Juan Pablo Tosso and the OWASP Coraza contributors.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package coraza
+
+import (
+	"fmt"
+	"io"
+	"net"
+	"net/http"
+	"strconv"
+	"strings"
+
+	"github.com/corazawaf/coraza/v3/types"
+)
+
+func processRequest(tx types.Transaction, r *http.Request) (*types.Interruption, error) {
```

Comment:
What differs here vs the coraza http middleware?

Reply:
The only difference is it takes the transaction as a parameter, so we can control the response